app/furniture/marketplace/carts/_cart.html.erb

```diff
@@ -1,20 +1,25 @@
 <div id="<%= dom_id(cart) %>">
 
-  <%- if cart.delivery_address.blank? %>
+  <%- if cart.delivery_address.blank? || cart.contact_phone_number.blank? %>
     <%= form_with model: cart.location do |cart_form| %>
       <%= render "text_field", attribute: :delivery_address, form: cart_form %>
+      <%= render "text_field", attribute: :contact_phone_number, form: cart_form %>
 
-      <%= cart_form.submit t('marketplace.delivery_address.edit') %>
+      <%= cart_form.submit t('marketplace.edit_delivery_info.edit') %>
```

Comment:
Nit: `marketplace.delivery_info.edit` is less duplicaive than `marketplace.edit_delivery_info.edit`

Reply:
good catch, I missed that. thanks. fixed in 2e304dfe65982b217c7aea7889a44d984214f9c3